alleycats-core/src/main/scala/alleycats/Empty.scala

```diff
@@ -77,7 +77,10 @@ object Empty extends EmptyInstances0 {
 
 }
 
-private[alleycats] trait EmptyInstances0 extends compat.IterableEmptyInstance with EmptyInstances1
+private[alleycats] trait EmptyInstances0 extends compat.IterableEmptyInstance with EmptyInstances1 {
+  private[this] val emptyOptionSingleton: Empty[Option[Any]] = Empty(None)
```

Comment:
I'd rather `Option[Nothing]` which is a more precise type.

Reply:
There are a number of cases in Cats where we use this trick with `Any` but not `Nothing`. However, honestly, I don't see any significant difference. I made it `Nothing` if it doesn't break existing uniformity.